SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -651,3 +660,115 @@ void ASpatialFunctionalTest::ClearTagDelegationAndInterest()
 		NetDriver->DebugCtx->Reset();
 	}
 }
+
+void ASpatialFunctionalTest::TakeSnapshot(const FSnapshotTakenDelegate& BlueprintCallback)
+{
+	TakeSnapshot(BlueprintCallback, nullptr);
+}
+
+void ASpatialFunctionalTest::TakeSnapshot(const FSnapshotTakenFunc& CppCallback)
+{
+	TakeSnapshot(FSnapshotTakenDelegate(), CppCallback);
+}
+
+void ASpatialFunctionalTest::TakeSnapshot(const FSnapshotTakenDelegate& BlueprintCallback, const FSnapshotTakenFunc& CppCallback)
+{
+	FHttpModule& HttpModule = FModuleManager::LoadModuleChecked<FHttpModule>("HTTP");
+	TSharedRef<class IHttpRequest> HttpRequest = HttpModule.Get().CreateRequest();
+	FString KrakenSnapshotURL = "http://localhost:31000/improbable.platform.runtime.SnapshotService/TakeSnapshot";
+	// FString SquidSnapshotUrl = "http://localhost:5006/snapshot";
+	HttpRequest->OnProcessRequestComplete().BindLambda([this, BlueprintCallback, CppCallback](
+														   FHttpRequestPtr HttpRequest, FHttpResponsePtr HttpResponse, bool bSucceeded) {
+		if (!bSucceeded)
+		{
+			UE_LOG(LogSpatialGDKFunctionalTests, Error, TEXT("Failed to trigger snapshot at '%s'; received '%s'"), *HttpRequest->GetURL(),
+				   *HttpResponse->GetContentAsString());
+			BlueprintCallback.ExecuteIfBound(false);
+			if (CppCallback != nullptr)
+			{
+				CppCallback(false);
+			}
+			return;
+		}
+
+		// Unfortunately by the time this callback happens, the files haven't been flushed, so if you copy you may get
+		// the wrong info! So let's wait a bit..
```

Comment:
:(

Reply:
Perhaps worth setting up a file watcher instead?

Reply:
This is also a problem with Squid btw, but I talked with Sasha to see if we can get a response only once the snapshot has been saved.

Reply:
What did he say?

Reply:
He said he'd bring up my requests, I'll follow up to get case numbers

Reply:
For reference: https://improbableio.atlassian.net/browse/SQD-871